src/fluree/db/util/context.cljc

```diff
@@ -43,26 +41,66 @@
                        :error  :db/invalid-context})))))
 
 
+(defn stringify-context-key
+  [k]
+  (cond
+    (string? k)
+    k
+
+    (keyword? k)
+    (if-let [ns (namespace k)]
+      (str ns ":" (name k))
+      (name k))
+
+    :else
+    (throw (ex-info (str "Context key appears to be invalid: " k)
+                    {:status 400
+                     :error  :db/invalid-context}))))
+
+(defn stringify-context-val
+  [v]
+  ;; reserved terms are not inclusive, only focused on possible map value terms (excludes e.g. @version)
+  (let [reserved-terms-map {:id        "@id"
+                            :type      "@type"
+                            :value     "@value"
+                            :list      "@list"
+                            :set       "@set"
+                            :context   "@context"
+                            :language  "@language"
+                            :reverse   "@reverse"
+                            :container "@container"
+                            :graph     "@graph"}]
+    (cond
+      (string? v)
+      v
+
+      (map? v)
+      (reduce-kv (fn [acc k' v']
+                   (let [k'* (or (get reserved-terms-map k')
+                                 (stringify-context-key k'))
+                         v'* (or (get reserved-terms-map v')
+                                 (stringify-context-key v'))]
```

Comment:
This is actually the code to specifically deal with that scenario. In your example, the input to this function would be `{"@container" "@set"}`. Because it is a map, it will hit the `(map? v)` condition, and the reduce-kv will mean `k'` = `@container` and `v'` = `@set`. Both will be sent to `stringify-context-key`, but because they are both already strings they will return unmodified.

Reply:
Alright, that makes sense, thanks for walking me through it.